protocol/implementations/js/src/builders.ts

```diff
@@ -0,0 +1,36 @@
+import { MessageTypes, TbDEXMessage } from './types.js'
+import {ulid} from 'ulidx'
+
+// maybe extract this somewhere?
+const ID_NAMESPACE = 'tbdex'
+
+function createId(messageType: string): string {
+  return `${ID_NAMESPACE}:${messageType}:${ulid()}`
+}
+
+export function createMessage<Type extends keyof MessageTypes>(to: string, from: string, body: MessageTypes[Type], messageType: Type,): TbDEXMessage<Type> {
```

Comment:
Is there a way to avoid having to pass in `messageType` again and to grab it from the generic?

Reply:
i wish! all types are erased during compilation